Stage 1.1 and Stage 1.2

mgildenberg opened the pull request:
**Overview**

This is the frontend code of an application that allows users to add entries (Slaylets) to a topic list (Slaylist), with notes to explain why they're adding it. Example use: a user can create a Slaylist of "Top Guitarists of Youtube" and link to 10 different Youtube channels, describing why each one deserves the spot.

First scope will be limited to Youtube account lists. Expansion ideas based on the native curation lacking notes or lacking any native option to curate that individual item.

- Instagram accounts
- TikTok accounts
- Spotify Albums
- individual Youtube videos (playlists lack notes)
- individual Instagram posts (saved collections lack notes)

**Keep In Mind**

Not everything is meant to be used in full in this frontend version right now. Examples:
- Because there is no backend, I am using a default set of data from the contants.js file
- You can register/login with any credentials that meet the form field requirements because there is already a default username from constants.js for the current user. 
- You can Like a SlaylistCard while logged in only and it will communicate up to the SlaylistModal that this content is Liked but doing the same thing on a SlaylistModal does not communicate back to SlaylistCard
- You can add subentries (Slaylets) on the New Slaylist form page but no Youtube API request will be made. Only your Title, Tagline, and Category are shown in a temporary SlaylistModal (along with default Slaylet data). In the future, this will change and there will be limits to the # of Slaylets.
- You can click into the Category on the NewSlaylist form to change from default Youtube to the Test one but right now it's not consequential which is used

ImaliaNsk commented:
Thanks for good work! Your project got enough points to be accepted. Congratulations, you did a great job. I wish you success on the next stage!

However, there are some bugs that might need fixing, please find the checklist and comment below.

# Summary:
### Performance Criteria

#### Markup and JSX
- [ ] <!--7.4--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions. There is no horizontal scrolling on resolutions of `320px` and up. The scrollbar can't be hidden using the overflow: hidden declaration.

* Home, dashboard, new-slaylist scroll horizontally on 322 - 350
* On some mobile resolutions, header elements don't fit inside their container: https://prnt.sc/Da83cJGnQVvz

- [ ] <!--7.26--> Navigation between pages and links to external resources are working correctly. There are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and there are no hidden blocks.

* Please note that the link in README leads to route `/slaylist-frontend` which is 404, not found.

- [x] <!--7.26--> Elements specified in the UI toolkit are used correctly (e.g., buttons, forms, text elements, etc).
- [x] <!--3.63--> Classes are named according to BEM specifications.
- [x] <!--3.63--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--3.63--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--3.63--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--3.63--> Infrastructural project files are created using CRA.
- The markup has been converted into JSX:
  - [x] <!--3.63--> The markup is included within parentheses `( )`.
  - [x] <!--3.63--> The markup has been moved into the corresponding components.
- [x] <!--3.63--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--3.63--> There are no linting errors or warnings.

#### React
- React components:
  - [x] <!--3.63--> Hooks are not used inside conditional statements or loops.
  - [x] <!--3.63--> Hooks are called in a component's main function.
  - [x] <!--3.63--> For class components, effects are described inside the component lifecycle methods.
- [x] <!--7.26--> Asynchronous API requests:
  - Requests can be made through the Fetch API or by using XMLHttpRequest. Third-party libraries (such as axios or jQuery) are not used.
  - API requests are contained in a separate file.
  - The chain for processing promises ends with a `catch()` block.
  - The first `then()` handler returns `res.json`.
- [x] <!--3.63--> Naming:
  - camelCase is used for function and variable names.
  - Only nouns are used as variable names.
  - Variable names clearly describe what is stored in them. If the project has several variables with similar data, then those variables have unique but descriptive names.
  - Descriptive names are used for functions, which reflect what they do.
  - Function names start with a verb.
  - JS classes and functional components are named using nouns and start with a capital letter.
  - Names must not include inappropriate or unclear abbreviations.
  - Сustom hook names start with `use`.
- [x] <!--3.63--> No third-party JavaScript libraries are used, unless absolutely necessary. If third-party libraries are connected, then they are used appropriately.

### Best Practices

#### Markup and JSX
- [x] <!--1.25--> The design reuses components wherever possible.
- [x] <!--1.25--> Fonts are connected using `@font-face`.
- [x] <!--1.25--> The icons are exported in SVG format.
- [ ] <!--1.25--> Form elements should be highlighted when focused.
- [x] <!--1.25--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--1.25--> The use of `reset.css` is prohibited.

#### React
- [x] <!--1.25--> The initial state of state variables contain the correct data type.
- [ ] <!--1.25--> API requests are described inside the `App` component.
- [x] <!--1.25--> There is no memory leak when hanging handlers. All handlers added with `addEventListener` are removed when the component is unmounted.
- [ ] <!--1.25--> API error handling:
  - [ ] The user receives a message in the case of an error.
- [x] <!--1.25--> Non-variable values (hard-coded constants) are named in all capital letters and stored in a separate configuration file.
- [x] <!--1.25--> If you are using popups, then correct the opening and closing functionality is implemented. Popups can be closed using the cross button, overlay, or ESC key.

### Recommendations

#### Markup and JSX
- [x] <!--0.83--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--0.83--> Images have an `alt` attribute containing appropriate values.
- [ ] <!--0.83--> Raster and vector images have been optimized.

#### React
- [x] <!--0.83--> For internal links in the application, components from the react-router library are used.
- [ ] <!--0.83--> Semantically correct blocks are used for components. No `<div />` or other unnecessary HTML tags are used for components that consist of single-level blocks.

* Slaylist items contain two unnecessary wrappers: `slaylist-card__content` and `slaylist-card__flex-container`, all necessary padding and flexbox rules can be assigned for `slaylist-card`
* `footer` contains two unnecessary div wrappers too.

- [x] <!--0.85--> The code is clean and easy to understand:
  - The code is readable and clearly structured. Some parts of the code are explained with comments if needed.
  - There is no extra code: for example, when a variable is declared but not used, or there is some kind of redundant logic.
  - The code is formatted in the same way, and the indentation hierarchy is respected.

## Number of points: 80

mgildenberg commented:
> - On some mobile resolutions, header elements don't fit inside their container: https://prnt.sc/Da83cJGnQVvz

Fixed

> Navigation between pages and links to external resources are working correctly. There are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and there are no hidden blocks.
> Please note that the link in README leads to route /slaylist-frontend which is 404, not found.

Fixed


>  API error handling:
>  The user receives a message in the case of an error.

I don't think this is relevant for stage 1.1/1.2 of this Slaylist project given there's no opportunity for the user to make an error (for now). My constants.js file is supplying valid Youtube handles and the API works as-is. An error right now would only occur in limited cases: the Youtube channel changes their handle, Youtube API changes, or my API key expires/changes permissions.

> API requests are described inside the App component.


This is not necessary. The YoutubeApi.js file's functions are called in the Slaylet.js component. I will likely change where it occurs, but even then I don't think anticipate needing it in App.js. 

-------

The other code feedback will be addressed.
